modules/json/src/smithy4s/http/json/SchemaVisitorJCodec.scala

```diff
@@ -1122,58 +1121,49 @@ private[smithy4s] class SchemaVisitorJCodec(maxArity: Int)
         out.writeObjectEnd()
     }
 
-    nonPayloadStruct(fields, maybeInputOutput)(make, encode)
+    nonPayloadStruct(fields, structHints)(make, encode)
   }
   override def struct[S](
       shapeId: ShapeId,
       hints: Hints,
       fields: Vector[SchemaField[S, _]],
       make: IndexedSeq[Any] => S
   ): JCodec[S] = {
-    (
```

Comment:
@daddykotex I've taken advantage of having to change this bit of code in order to reduce the nesting of the various pattern matches

Reply:
that's great

Reply:
these are a few things I'll have to backport to my PR

Reply:
your PR has been merged

Reply:
lol just realized that as I was looking at the PR (did not do a round of notification lol)